### PR TITLE
Store logs from master components from Kubemark runs in GCS

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #21254

### CODE REVIEW

File: test/e2e/core.go [hunk 1]
```diff
@@ -37,55 +37,61 @@ func CoreDump(dir string) {
...
-	if !providerIs(providersWithSSH...) {
+	// requires ssh to master machine
+	if !providerIs(providersWithMasterSSH...) {
 		fmt.Printf("Skipping SSH core dump, which is not implemented for %s", provider)
 		return
 	}
 
-	// Get all nodes' external IPs.
-	hosts, err := NodeSSHHosts(c)
+	// I wish there was a better way to get the master IP...
```
Comment (wojtek-t) [inline test/e2e/core.go]: FYI - this is just changing the order. We first copy files from master machine and only then from nodes (it was the oposite order before).

**Comment by wojtek-t** [comment]
Ref #20638

File: test/e2e/util.go [hunk 2]
```diff
@@ -314,6 +314,9 @@ func SkipUnlessServerVersionGTE(v semver.Version, c client.ServerVersionInterfac
 // providersWithSSH are those providers where each node is accessible with SSH
 var providersWithSSH = []string{"gce", "gke", "aws"}
 
+// providersWithMasterSSH are those providers where master node is accessible with SSH
+var providersWithMasterSSH = []string{"gce", "gke", "kubemark", "aws"}
```
Comment (ixdy) [inline test/e2e/util.go]: `gke` should not have been in this list.

Comment (wojtek-t) [inline test/e2e/util.go]: So why it is with "providerswithSSH"? Previously we were using it also to access master.

Comment (ixdy) [inline test/e2e/util.go]: it previously didn't differentiate between nodes and masters, but it probably should have. ssh to GKE masters has always been disallowed.

**Pull request merged**